src/extension.ts

```diff
@@ -30,29 +55,157 @@ class DDBViewProvider implements vscode.WebviewViewProvider {
             enableScripts: true,
             localResourceRoots: [this._extensionUri]
         };
-        webviewView.webview.html = this._getHtmlForWebview(webviewView.webview);
+        webviewView.webview.onDidReceiveMessage(
+            message => {
+                switch (message.command) {
+                    case 'clear_messages':
+                        gpt_messages_array = [];
+                        md_messages_array = [];
+                        return;
+
+                    case 'get_gpt_response':
+                        this.getGptResponse(message.id, message.content);
+                        return;
+
+                    case 'restore_messages':
+                        gpt_messages_array = message.content;
+                        return;
+                }
+            },
+            undefined,
+            this.context.subscriptions
+        );
+        webviewView.webview.html = this.getHtmlForWebview(webviewView.webview);
         this.webViewGlobal = webviewView;
     }
 
-    private _getHtmlForWebview(webview: vscode.Webview) {
+    public async addMessageToChat(message: string, prompt: string) {
+        await vscode.commands.executeCommand('ddb50.chatWindow.focus').then(() => {
+            setTimeout(() => {
+                this.webViewGlobal!.webview.postMessage(
+                    {
+                        command: 'addMessage',
+                        content: {
+                            "userMessage": message,
+                        }
+                    });
+                this.getGptResponse(uuid.v4(), prompt, false);
+            }, 100);
+        });
+    }
+
+    private getGptResponse(id: string, content: string, persist_messages = true) {
+
+        try {
+            if (persist_messages) {
+                gpt_messages_array.push({ role: 'user', content: content });
+                md_messages_array.push({ role: 'user', content: content});
+                this.webViewGlobal!.webview.postMessage(
+                    {
+                        command: 'persist_messages',
+                        gpt_messages_array: gpt_messages_array,
+                        md_messages_array: md_messages_array
+                    }
+                );
+            }
+
+            const postOptions = {
+                method: 'POST',
+                host: 'cs50.ai',
+                port: 443,
+                path: '/api/v1/ddb50',
+                headers: {
+                    'Authorization': `Bearer ${process.env['CS50_TOKEN']}`,
```

Comment:
If a user adds a `\n` or some other character to their `CS50_TOKEN`, could this break the HTTP call, such that we should somehow escape the value?

Reply:
Sure, we could just .trim() on the client side and also .strip() on the server-side.

Reply:
Able to check if `https.requests` already escapes HTTP headers for callers? Technically, `trim` alone wouldn't be enough; we'd ideally escape the values according to what's allowed (e.g., printable ASCII characters, no control characters, etc.). Though I imagine `https.requests` might do that for us?

No need for `strip` on server side; by the time the headers reach the server, any newlines or such would likely make the server think multiple values had been received.

Reply:
But why would `CS50_TOKEN` had these characters?

Reply:
Just principle, coming from user's environment, so value might not be as we expect.

Reply:
`http.requests` don't automatically escape HTTP headers, because HTTP headers are not supposed to contain characters that need to be escaped. We could still do sanity check regardless, though I don't see a scenario of why `CS50_TOKEN` would have those characters if we were obtaining the token from oauth?